specs/CookieManagement.md

```diff
@@ -0,0 +1,507 @@
+                {
+                    message = message.substr(10);
+                    std::wstring name = message.substr(0, message.find(' '));
+                    std::wstring value = message.substr(message.find(' ') + 1);
+
+                    //! [AddOrUpdateCookie]
+                    wil::com_ptr<ICoreWebView2Cookie> cookie;
+                    CHECK_FAILURE(m_cookieManager->CreateCookie(
+                        name.c_str(), value.c_str(), L".bing.com", L"/", &cookie));
+                    CHECK_FAILURE(m_cookieManager->AddOrUpdateCookie(cookie.get()));
+                    reply = L"{\"CookieAddedOrUpdated\":\"Cookie added or updated successfully.\"}";
+                    CHECK_FAILURE(sender->PostWebMessageAsJson(reply.c_str()));
+                    //! [AddOrUpdateCookie]
+                }
+                else if (message.compare(0, 16, L"ClearAllCookies ") == 0)
+                {
+                    CHECK_FAILURE(m_cookieManager->DeleteAllCookies());
+                    reply = L"{\"CookiesDeleted\":\"Cookies all deleted.\"}";
+                    CHECK_FAILURE(sender->PostWebMessageAsJson(reply.c_str()));
+                }
+                return S_OK;
+            })
+            .Get(),
+        &m_webMessageReceivedToken));
+}
+
+static std::wstring CookieToJsonString(ICoreWebView2Cookie* cookie)
+{
+    //! [CookieObject]
+    wil::unique_cotaskmem_string name;
+    CHECK_FAILURE(cookie->get_Name(&name));
+    wil::unique_cotaskmem_string value;
+    CHECK_FAILURE(cookie->get_Value(&value));
+    wil::unique_cotaskmem_string domain;
+    CHECK_FAILURE(cookie->get_Domain(&domain));
+    wil::unique_cotaskmem_string path;
+    CHECK_FAILURE(cookie->get_Path(&path));
+    double expires;
+    CHECK_FAILURE(cookie->get_Expires(&expires));
+    BOOL httpOnly;
+    CHECK_FAILURE(cookie->get_IsHttpOnly(&httpOnly));
+    COREWEBVIEW2_COOKIE_SAME_SITE_KIND same_site;
+    std::wstring same_site_as_string;
+    CHECK_FAILURE(cookie->get_SameSite(&same_site));
+    switch (same_site)
+    {
+    case COREWEBVIEW2_COOKIE_SAME_SITE_KIND_NONE:
+        same_site_as_string = L"None";
+        break;
+    case COREWEBVIEW2_COOKIE_SAME_SITE_KIND_LAX:
+        same_site_as_string = L"Lax";
+        break;
+    case COREWEBVIEW2_COOKIE_SAME_SITE_KIND_STRICT:
+        same_site_as_string = L"Strict";
+        break;
+    }
+    BOOL secure;
+    CHECK_FAILURE(cookie->get_IsSecure(&secure));
+
+    std::wstring result = L"{";
+    result += L"\"Name\": " + EncodeQuote(name.get()) + L", " + L"\"Value\": " +
+              EncodeQuote(value.get()) + L", " + L"\"Domain\": " + EncodeQuote(domain.get()) +
+              L", " + L"\"Path\": " + EncodeQuote(path.get()) + L", " + L"\"HttpOnly\": " +
+              BoolToString(httpOnly) + L", " + L"\"Secure\": " + BoolToString(secure) + L", " +
+              L"\"SameSite\": " + EncodeQuote(same_site_as_string) + L", " + L"\"Expires\": ";
+    if (expires == -1)

```

Comment:
Can we make an IsSession property for the cookie so we don't have to have a special value for the expires? This will also help for .NET and WinRT where we'll use a date type and won't have an easy way to check the expires.